### DISCUSSION TITLE
Prioritized NorESM2.5 twin simulations  - July 2024

### DISCUSSION THREAD

**Discussion by adagj** [NorESMhub/noresm3_dev_simulations#7] [Discussions about coupled simulations]
### Simulation: Trigrid
Description: Test the impact of running the land component on FV grid instead of SE grid. This simulation will run atmosphere (SE) and land (FV) on different grids
Set up: N1850, fully coupled with Oslo-Aero, CAM on SE grid, CLM on FV grid,  BLOM (hybrid) and CICE on tnx14
Duration: 10 years
Responsible: Mariana

### Simulation: Dual grid
Description: Test the impact of running the land component on FV grid instead of SE grid. This simulation will run atmosphere and land on the same spectral element grid
Set up: N1850, fully coupled with Oslo-Aero, CAM and CLM on SE grid, CAM6 radiation, BLOM (hybrid) and CICE on tnx14
Duration: 10 years 
Responsible: Mariana

### Simulation: Hybrid BLOM
Description: Test the impact of the new vertical grids in BLOM. This simulation will run BLOM with the hybrid coordinate grid
Set up: N1850, fully coupled with Oslo-Aero CAM on SE grid, CLM on FV grid, CAM6 radiation, BLOM (hybrid) and CICE on tnx14
Duration: 10 years
Responsible: Ada

### Simulation : Isopycnic BLOM
Description: Test the impact of different vertical ocean grids. This simulation will run BLOM with the isopycnic vertical grid (used in CMIP6) 
Set up: N1850, fully coupled with Oslo-Aero, CAM on SE grid, CLM on FV grid, CAM6 radiation, BLOM (isopycnic) and CICE on tnx14 
Duration: 10 years
Responsible: Ada

### Simulation: Oslo-Aero
Description: Test full NorESM configuration with trigrid, hybrid BLOM, OSLO-AERO
Set up: N1850, fully coupled with Oslo-Aero, CAM6 radiation, dms from ocean, CAM on SE grid, CLM on FV grid, BLOM (hybrid) and CICE on tnx14
Duration: 10 years 
Responsible: Øyvind

### Simulation: Oslo-Aero-COARE
Description: Test full NorESM configuration with trigrid, hybrid BLOM, and OSLO-AERO, and COARE flux parameterization
Set up: N1850, fully coupled with Oslo-Aero, CAM6 radiation, dms from ocean, COARE flux param., CAM on SE grid, CLM on FV grid, BLOM (hybrid) and CICE on tnx14 
Duration: 10 years 
Responsible: Øyvind

**Comment by oyvindseland** [comment]
Code version for simulation ?
https://github.com/mvertens/NorESM/tree/feature/noresm2_5_alpha03_v1 ?

**Reply by mvertens** [reply]
@oyvindseland - I need to create a new alpha version - noresm2_5_alpha03_v2. However I need to test this on betzy before I create it - and betzy is still not functional. As soon as betzy is working - I'll start the testing and let everyone know.

**Reply by oyvindseland** [reply]
OK: I see Betzy is up but with quite limited access to the compute cores

**Comment by oyvindseland** [comment]
How should we give the trigrid as a shortname in case name?
ne30_f19_tn14 ?

**Reply by mvertens** [reply]
I have created a new alias ne30_f09_tn14 - but it will only be available in noresm2_3_alpha03_v2.

**Reply by oyvindseland** [reply]
Sorry, I wrote a 2 degree definition. Yes so ne30_f09_tn14

**Comment by mvertens** [comment]
@oyvindseland @adagj @matsbn - where should the cases for these simulations be create?

**Reply by adagj** [reply]
We already discussed this during the core meeting, and agreed that the cases should not be stored in a common directory. However, the case folders need to be archived together with the hist files and all output in the common directory /projects/NS9560K/noresm3/cases/ which I will make now

**Reply by adagj** [reply]
@oyvindseland @mvertens Can you please test if you have write access 
@NIRD /projects/NS9560K/noresm3/cases

**Comment by oyvindseland** [comment]
I think a common case directory for the test simulations can be a bit crowded so I suggest to do the simulations on /cluster/projects/nn9560k/$user/ 
I prefer the structure /cluster/projects/nn9560k/$user/nameofnoresmversion/cases/ myself but can also put it directly under /cluster/projects/nn9560k/$user/cases/ Since we now agreed on to put version name in the compset, information about the version is visible anyway.

On nird however I think we should have the archive on a common directory. /projects/NS9560K/noresm2.5/cases/ ?

**Reply by mvertens** [reply]
@oyvindseland - my experience with CESM is that having a common directory of simulations was extremely helpful in tracking things. Maybe with this new database its not as important. However we would have to know where to look under each user to see the right nameofnoresm. I would propose that if we want to keep users separate we use
  /cluster/projects/nn9560k/nameofversion/$user/nameofnoresm - so that at least all experiments for a given noresm are grouped under one directory but separated by users.

**Reply by mvertens** [reply]
@adagj @matsbn @MichaelSchulzMETNO @oyvindseland - it looks like the trigrid simulation done was for oslo-aero not for ghgmam4. I am wondering at this point if we should not target the dual/trigrid twin simulations to have oslo-aero as opposed to mam4 for the full coupled configurations. That would be a lot closer to what we really want to use.

**Reply by oyvindseland** [reply]
Are you now referring to test_nf1850oslo_trigrid or a new coupled with the settings as defined for simulation set 3 above?

**Reply by mvertens** [reply]
I'm referring to the dualgrid/trigrid fully coupled twin simulations that I will be doing. It specifies mam4 - but at this point I'm wondering if it should be oslo-aero - since our previous dualgrid/trigrid comparison was also oslo-aero but for an amip configuration.

**Reply by oyvindseland** [reply]
I think that it is best to focus on oslo-aero simulations since the impact may be different on mam4 and analysing maa4 should not be our main target but for this I hope that Ada and Michael also  add to the discussion.

**Reply by adagj** [reply]
I agree. The reason was that oslo aero was not possible to run with the most recent radiation code, at least that is what we thought. Since the default CAM configuration so far contains the "old" radiation code, I see no reason to use MAM4 as default. I can update the description of the twin simulations accordingly. However, the most important aspect of the twin simulations is to only change one thing at the time.

**Reply by adagj** [reply]
Let's use oslo aero for all the twin simulations. But will simulation 1 in set 1 and simulation 1 in set 3 be identical? Then we only need one of them

**Reply by oyvindseland** [reply]
That may even be true for simulation  1 in set 3. Unless set 3 is the only one with DMS from ocean?

**Comment by oyvindseland** [comment]
Separating out the discussion on the placement of the experiments. 
I can certainly live with 
cluster/projects/nn9560k/nameofversion/$user/nameofnoresm
but if so would the nameofversion stay the same for all test experiments leading to individual pull requests until a new version / tag is defined?

**Reply by adagj** [reply]
I suggest to start a new discussion for this topic

**Reply by oyvindseland** [reply]
I agree with this.

**Comment by MichaelSchulzMETNO** [comment]
I think the reason to use MAM4 was that for the latest CAM version only MAM4 was working. But now that the new radiation is not any more an argument, I dont see why the MAM4 should be used throughout, if OSLO-Aer could be used instead. Are there other details of CAM which are newer when including MAM4?

**Reply by adagj** [reply]
I agree. The reason was that oslo aero was not possible to run with the most recent radiation code, at least that is what we thought. Since the default CAM configuration so far contains the "old" radiation code, I see no reason to use MAM4 as default. I can update the description of the twin simulations accordingly. However, the most important aspect of the twin simulations is to only change one thing at the time.

**Reply by mvertens** [reply]
So for the dualgrid/trigrid comparison - both will use oslo-aero in a fully coupled configuration using the same compset. The only thing that will change is the grid.

**Comment by adagj** [comment]
I suggest in the future that we make a separate discussion for each topic under discussion. This thread has become messy. As an example, for this particular discussion thread, I thing it would be better to have separate discussion threads for: case name (which short name to use), common case folder or not, where to store the output. While if Oslo-Aero should be used as default or not belong in this thread. 
@oyvindseland @mvertens

**Comment by oyvindseland** [comment]
The update on Betzy have run into trouble and may not be complete before end of next week.  https://opslog.sigma2.no/
Presumably it will work all the time but with variable capacity

The capacity today are only 20 % so if it will be hard to get the job running this week if the situation does not improve soon.
https://www.sigma2.no/hardware-status

 If the August update runs into similar type of problems I guess we should consider if we can do some development work on Fram
